sdks/java/io/amazon-web-services2/src/test/java/org/apache/beam/sdk/io/aws2/s3/SSECustomerKeyTest.java

```diff
@@ -54,4 +55,18 @@ public void buildWithArgs(String key, String algorithm, String md5, String encod
     assertEquals(algorithm, sseCustomerKey.getAlgorithm());
     assertEquals(encodedMD5, sseCustomerKey.getMD5());
   }
+
+  @Test
+  public void testJsonSerializeDeserialize() {
+    // default key created by S3Options.SSECustomerKeyFactory
+    SSECustomerKey emptyKey = SSECustomerKey.builder().build();
+    assertThat(jsonSerializeDeserialize(emptyKey)).isEqualToComparingFieldByField(emptyKey);
```

Comment:
I know you just moved the test method, but why is this needed to use `isEqualToComparingFieldByField` rather than `isEqualTo` in that case ?

Reply:
`SSECustomerKey` lacks an implementation of equals ....